src/integ_test/java/games/strategy/engine/chat/ChatIntegrationTest.java

```diff
@@ -79,28 +81,30 @@ public void tearDown() {
     }
   }
 
-  @Test(timeout = 15_000)
-  public void shouldBeAbleToChatAcrossMultipleNodes() throws Exception {
-    final ChatController controller = newChatController();
-    final Chat server = newChat(serverMessenger, serverChannelMessenger, serverRemoteMessenger);
-    server.addChatListener(serverChatListener);
-    final Chat client1 = newChat(client1Messenger, client1ChannelMessenger, client1RemoteMessenger);
-    client1.addChatListener(client1ChatListener);
-    final Chat client2 = newChat(client2Messenger, client2ChannelMessenger, client2RemoteMessenger);
-    client2.addChatListener(client2ChatListener);
-    waitFor(this::allNodesToConnect);
-
-    sendMessagesFrom(client2);
-    sendMessagesFrom(server);
-    sendMessagesFrom(client1);
-    waitFor(this::allMessagesToArrive);
-
-    client1.shutdown();
-    client2.shutdown();
-    waitFor(this::clientNodesToDisconnect);
-
-    controller.deactivate();
-    waitFor(this::serverNodeToDisconnect);
+  @Test
+  public void shouldBeAbleToChatAcrossMultipleNodes() {
+    assertTimeout(Duration.ofSeconds(15), () -> {
```

Comment:
There's a assertTimeoutPreemptively method, which fails instantly after the timeout limit is reached.
This method always lets the test methods finish first. Should we use the other method?

Reply:
I just learned that this is the Junit4 behaviour... Changing that now
https://github.com/junit-team/junit4/wiki/Timeout-for-tests

Reply:
Agreed,  It looks like the rule-of-thumb is to use `assertTimeout()` for things like performance tests (e.g. this code that never blocks needs to run within a certain amount of time), while `assertTimeoutPreemptively()` should be used for anything that can possibly block.